src/components/flash/flash.js

```diff
@@ -133,9 +135,16 @@ class Flash extends React.Component {
    * @method componentWillReceiveProps
    * @return(Void)
    */
-  componentWillReceiveProps(prevProps) {
-    if (prevProps.open !== this.props.open) {
-      this.setState({ dialogs: {} });
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open === this.props.open) { return; }
+
+    if (nextProps.open) {
+      this.setState({ dialogs: {}, open: nextProps.open });
+    } else {
+      clearTimeout(this.fadeLeaveTimeout);
+      this.fadeLeaveTimeout = setTimeout(() => {
+        this.setState({ dialogs: {}, open: nextProps.open });
```

Comment:
same again, don't think you need to set `dialogs`?

Reply:
Here we don't need it here for sure.